workshops/models.py

```diff
@@ -1950,3 +1993,152 @@ class TrainingRequest(ActiveMixin, CreatedUpdatedMixin, models.Model):
         default='', null=False, blank=True,
         help_text='What else do you want us to know?',
         verbose_name='Anything else?')
+
+    STATES = (
+        ('p', 'Pending'),  # initial state
+        ('a', 'Accepted'),  # state after matching a Person record
+        ('d', 'Discarded'),  # FIXME or deleted?
+    )
+    state = models.CharField(choices=STATES, default='p', max_length=1)
+
+    person = models.ForeignKey(Person, null=True, blank=True)
+
+    # no association with Event
+
+    @property
+    def state_label(self):
+        """Used in templates."""
+
+        switch = {
+            'p': 'warning',
+            'a': 'success',
+            'd': 'danger',
+        }
+        return switch[self.state]
+
+    def clean(self):
+        super().clean()
+
+        if self.state == 'a' and self.person is None:
+            raise ValidationError({'person': 'Accepted training request must '
+                                             'be matched to a person.'})
+
+    def get_absolute_url(self):
+        return reverse('trainingrequest_details', args=[self.pk])
+
+
+class TrainingRequirement(models.Model):
+    name = models.CharField(max_length=STR_MED)
+
+    # Determines whether TrainingProgress.url is required (True) or must be
+    # null (False).
+    url_required = models.BooleanField(default=False)
+
+    # Determines whether TrainingProgress.event is required (True) or must be
+    # null (False).
+    event_required = models.BooleanField(default=False)
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        ordering = ['name']
+
+
+@reversion.register
+class TrainingProgress(CreatedUpdatedMixin, models.Model):
+    trainee = models.ForeignKey(Person, on_delete=models.PROTECT)
+
+    # Mentor/examiner who evaluates homework / session. May be null when a
+    # trainee submits their homework.
+
+    # FIXME Maybe let's rename it as evaluated_by?
+    awarded_by = models.ForeignKey(Person,
+                                   on_delete=models.PROTECT,
+                                   null=True, blank=True,
+                                   related_name='+')
+    requirement = models.ForeignKey(TrainingRequirement,
+                                    on_delete=models.PROTECT)
+
+    STATES = (
+        ('n', 'Not-evaluated yet'),
+        ('f', 'Failed'),
+        ('p', 'Passed'),
+    )
+    state = models.CharField(choices=STATES, default='p', max_length=1)
+
+    # When we end training and trainee has gone silent, or passed their
+    # deadline, we set this field to True.
+    discarded = models.BooleanField(
+        default=False,
+        verbose_name='Discarded -- check if the trainee has gone silent '
+                     'or passed their training deadline.')
+
+    event = models.ForeignKey(Event, null=True, blank=True)
```

Comment:
Just a thought: you can narrow choices here to only TTT events with `limit_choices_to`.